src/keyfile.c

```diff
@@ -1356,14 +1356,17 @@ static gboolean open_session_file(gchar **tmp, guint len)
  * for all files opened within this function */
 void configuration_open_files(GPtrArray *session_files)
 {
-	gint i;
 	gboolean failure = FALSE;
+	const gboolean save_tab_order_beside = file_prefs.tab_order_beside;
+	const gboolean save_tab_order_ltr = file_prefs.tab_order_ltr;
 
 	/* necessary to set it to TRUE for project session support */
 	main_status.opening_session_files++;
 
-	i = file_prefs.tab_order_ltr ? 0 : (session_files->len - 1);
-	while (TRUE)
+	file_prefs.tab_order_beside = FALSE;
+	file_prefs.tab_order_ltr = TRUE;
```

Comment:
Why do you override the pref temporarily? Is there some nested function call that looks at the pref?

Reply:
Yes, `notebook_new_tab()`.  Call chain is `open_session_file() → document_open_file_full() → document_create() → notebook_new_tab()`.

Reply:
Then I think `notebook_new_tab()` should look at `main_status.opening_session_files`. This is the primary indication that we are in "batch open mode" if some code must behave differently. I also used that flag to open docs in a idle callback instead of synchronously (see #3267 / commit 23367de0c5)

Reply:
@kugel- OK, makes sense.  Do you like it better like that (updated)?

Reply:
yes (although I had a hard time to decode the new block in my brain)

Reply:
Yeah it required a bit of refactoring, because the previous code was a mess…

@elextr @eht16 @techee are you happy with this, and can you give this a new spin?

Reply:
Nice! It:
1. Fixes the issue of the reordered tabs
2. Also opens a file from the command-line as the last tab which is much more logical than the second tab before
3. Eliminates some ugly code

The code looks good to me as well.

Reply:
I had only a quick look on the code but tested it again and it works fine.